src/Widgets/AppButton.vala

```diff
@@ -140,10 +140,27 @@ public class Slingshot.Widgets.AppButton : Gtk.Button {
                 app_launched ();
             });
 
-            if (menu != null && menu.get_children () != null) {
+            if (menu.get_children ().length () > 0) {
```

Comment:
Actually, that check for `null` children was alright. If the list empty GLib treats this as a `NULL` pointer.
From the documentation (https://developer.gnome.org/glib/stable/glib-Doubly-Linked-Lists.html): 
```
There is no function to create a GList. NULL is considered to be a valid, empty list so you simply set a GList* to NULL to initialize it.
```

Reply:
Thanks for this.